hdkeychain/extendedkey.go

```diff
@@ -206,7 +206,11 @@ func (k *ExtendedKey) ParentFingerprint() uint32 {
 // index does not derive to a usable child.  The ErrInvalidChild error will be
 // returned if this should occur, and the caller is expected to ignore the
 // invalid child and simply increment to the next index.
-func (k *ExtendedKey) Child(i uint32) (*ExtendedKey, error) {
+//
+// Use the 2nd parameter to fix the bug with leading zeros. This change was made
+// as an optional parameter in order to maintain backward compatibility and avoid
+// losing wallets created before fixing this bug.
+func (k *ExtendedKey) Child(i uint32, fixLeadingZeroBug ...bool) (*ExtendedKey, error) {
```

Comment:
This solution strikes me as being...inelegant. With this for the foreseeable future, the API is forced to carry a parameter that conditionally fixes an issue of spec compliance. This isn't backwards compatible as the comment indicates since this is actually itself a breaking API change, so it warrants a `v2` major version bump IMO. 

Another option would be to add a _new_ method that always applies the leading zero fix, with the existing public one calling that with the fix toggled off.

Reply:
> This solution strikes me as being...inelegant. With this for the foreseeable future, the API is forced to carry a parameter that conditionally fixes an issue of spec compliance. This isn't backwards compatible as the comment indicates since this is actually itself a breaking API change, so it warrants a `v2` major version bump IMO.

Can you explain why this cannot be considered backward compatible? This parameter is optional and does not correct the default error if it is absent.

Reply:
If someone created an interface in their app under the assumption that ExtendedKey fulfills it, and their interface has the call signature of `Child(i uint32) (*TheirInterface, error)` suddenly ExtendedKey no longer fulfills it.

Therefore it is breaking.

Reply:
I've come around to this approach, if you can update this PR before #182 does, I think it can land.